test/core/security/grpc_tls_certificate_provider_test.cc

```diff
@@ -495,8 +495,47 @@ TEST_F(GrpcTlsCertificateProviderTest,
   CancelWatch(watcher_state_1);
 }
 
-}  // namespace testing
+TEST_F(GrpcTlsCertificateProviderTest, EmptyPrivateKeyString) {
+  absl::Status status =
+      PrivateKeyPublicKeyMatches(/*private_key_string=*/"", cert_chain_);
+  EXPECT_EQ(status.code(), absl::StatusCode::kInvalidArgument);
+  EXPECT_EQ(status.message(), "Private key string is empty.");
+}
+
+TEST_F(GrpcTlsCertificateProviderTest, EmptyCertificateString) {
+  absl::Status status =
+      PrivateKeyPublicKeyMatches(private_key_2_, /*cert_string=*/"");
+  EXPECT_EQ(status.code(), absl::StatusCode::kInvalidArgument);
+  EXPECT_EQ(status.message(), "Certificate string is empty.");
+}
 
+TEST_F(GrpcTlsCertificateProviderTest, FailedStringToX509Conversion) {
```

Comment:
When naming the test, try to describe the behavior that you are testing against. For example, this can be `FailedKeyCertMatchOnInvalidCertFormat`.

Reply:
Noted, thanks!